External graphs

erickt opened the pull request:
This PR adds support for external graphs. To create one, do:

```
curl \
  -XPOST \
  -H 'content-type: application/json' \
  /api/projects/<project-id>/graphs \
  -d '{
    "properties":{
      "storage.backend":"local",
      "storage.directory": "/tmp/foo"
    }
  }'
```

It also does some minor cleanup, like fixing a race on opening graphs, corrects a typo in our example graph creation script, and finally allows `dendrite run-groovy` to print out the result of the script.

Commit:
./bin/dendrite run-groovy should print the script's output

Commit:
fix a typo in init-graph-of-the-gods.groovy

Commit:
Synchronize on opening the Titan graphs

cglewis commented on src/main/java/org/lab41/dendrite/web/controller/GraphController.java:
```diff
@@ -4,6 +4,7 @@
 import org.lab41.dendrite.models.ProjectMetadata;
 import org.lab41.dendrite.services.MetadataService;
 import org.lab41.dendrite.services.MetadataTx;
+import org.lab41.dendrite.web.beans.CreateGraphBean;
```
This breaks the build.

Commit:
Add support for creating a graph that points at an external hbase table/es index

This is done by posting doing:

curl \
  -XPOST \
  -H 'content-type: application/json' \
  /api/projects/<project-id>/graphs \
  -d '{
    "properties":{
      "storage.backend":"local",
      "storage.directory": "/tmp/foo"
    }
  }'

cglewis commented:
LGTM.
